include/envoy/upstream/cluster_manager.h

```diff
@@ -150,6 +150,12 @@ class ClusterManager {
    */
   virtual const envoy::api::v2::core::BindConfig& bindConfig() const PURE;
 
+  /**
+   * @return const envoy::api::v2::UpstreamConnectionOptions& cluster manager wide connection
+   * options for new upstream connections.
+   */
+  virtual const envoy::api::v2::UpstreamConnectionOptions& connectionOptions() const PURE;
```

Comment:
Given that configs are usually machine-generated, should we remove this (and the corresponding data-plane-api field) and just set this on each cluster?

Reply:
This allows it to be set in the bootstrap config easily, and makes it's behaviour consistent with upstream_bind_options.